test/lib/TestDialect/TestDialect.cpp

```diff
@@ -41,6 +41,19 @@ struct TestOpAsmInterface : public OpAsmDialectInterface {
     if (auto asmOp = dyn_cast<AsmDialectInterfaceOp>(op))
       setNameFn(asmOp, "result");
   }
+
+  void getAsmBlockArgumentNames(Block *block,
+                                OpAsmSetValueNameFn setNameFn) const final {
+    auto op = block->getParentOp();
+    if (auto arrayAttr = op->getAttrOfType<ArrayAttr>("arg_names")) {
```

Comment:
Prefer early exit:

auto arrayAttr = ...;
if (!arrayAttr)
  return;

Reply:
Why?

Reply:
Same as the rest.